bin/migrate-oats-data/applications/decisions/app_components/lot/application_decision_component_lot_insert.py

```diff
@@ -0,0 +1,143 @@
+        ) as sql_file:
+            query = sql_file.read()
+            while True:
+                cursor.execute(
+                    f"""{query} 
+                    WHERE ospi.subdiv_parcel_intent_id > {last_parcel_intent_id} ORDER BY ospi.subdiv_parcel_intent_id;"""
+                )
+
+                rows = cursor.fetchmany(batch_size)
+
+                if not rows:
+                    break
+                try:
+                    lots_to_be_inserted_count = len(rows)
+
+                    _insert_application_component_lots(conn, batch_size, cursor, rows)
+
+                    successful_inserts_count = (
+                        successful_inserts_count + lots_to_be_inserted_count
+                    )
+                    last_parcel_intent_id = dict(rows[-1])["subdiv_parcel_intent_id"]
+
+                    logger.debug(
+                        f"retrieved/inserted items count: {lots_to_be_inserted_count}; total successfully inserted component_lots so far {successful_inserts_count}; last inserted parcel_intent_id: {last_parcel_intent_id}"
+                    )
+                except Exception as err:
+                    logger.exception(err)
+                    conn.rollback()
+                    failed_inserts = count_total - successful_inserts_count
+                    last_parcel_intent_id = last_parcel_intent_id + 1
+
+    logger.info(
+        f"Finished {etl_name}: total amount of successful inserts {successful_inserts_count}, total failed inserts {failed_inserts}"
+    )
+
+
+def _insert_application_component_lots(conn, batch_size, cursor, rows):
+    query = _get_insert_query()
+    parsed_data_list = _prepare_oats_component_lots_data(rows)
+
+    if len(parsed_data_list) > 0:
+        execute_batch(cursor, query, parsed_data_list, page_size=batch_size)
+
+    conn.commit()
+
+
+def _get_insert_query():
+    query = f"""
+                INSERT INTO alcs.application_decision_component_lot (
+                    size,
+                    type,
+                    audit_created_by,
+                    component_uuid,
+                    index
+                )
+                VALUES (
+                    %(parcel_area)s,
+                    %(type)s,
+                    %(audit_created_by)s,
+                    %(decision_component_uuid)s,
+                    %(index)s
+                )
+                ON CONFLICT DO NOTHING;
+    """
+    return query
+
+
+def _prepare_oats_component_lots_data(row_data_list):
+    data_list = []
+    component_id_counts ={}
+    for row in row_data_list:
+        mapped_row = {
+            "parcel_area": row.get("parcel_area"),
+            "decision_component_uuid": row.get("uuid"),
+            "audit_created_by": OATS_ETL_USER,
+            "index": _create_parcel_index(row, component_id_counts),
+            "type": "Lot",
+
+        }
+        data_list.append(mapped_row)
+
+    return data_list
+
+
+def _create_parcel_index(row, component_id_counts):
```

Comment:
question: Why is this called create_parcel_index if it returns a dictionary? I assumed it would create a DB index or similar.

Reply:
I think in this case, the index is 1,2,3.... values in index column of lot table, rather than DB index

Reply:
That is correct, there is no index in oats, this is for the column "index"

Reply:
Ahh that makes sense, forgot the column was called index!